common/os/Mutex.cxx

```diff
@@ -36,8 +36,10 @@ Mutex::Mutex()
 #else
   int ret;
 
+  pthread_mutexattr_init(&ma);
+  pthread_mutexattr_settype(&ma, PTHREAD_MUTEX_RECURSIVE);  // Windows CriticalSections are recursive too
```

Comment:
I would prefer if we don't rely on recursive behaviour as it is encourages poor tracking of where locks are actually used.

Reply:
Disagree. This change here is only for matching with other platforms. I would avoud different behaviour on different systems

Reply:
To make my point clearer: Any debug proccess rely not only on application code but also on operation system behviour. And when we are writing a crossplatform application we **should** expose any operation system primitives and functionality to our code in the same manner. Otherwise if we want to debug our application code we will try to understand OS-related differences everywhere in the code and logic too. I would prefer not to to like that. If it would be possible to make Windows critical section non-reenterant I would have fix critical section. But it seems that they cant be non-reenterant.

But if this consern don't bother's you, I will rewrite this piece of code.